src/Microsoft.Health.Fhir.SqlServer/Features/Watchdogs/DefragWatchdog.cs

```diff
@@ -272,85 +387,75 @@ private double GetNumberParameterById(string id)
 INSERT INTO dbo.Parameters (Id,Number) SELECT @ThreadsId, 4 WHERE NOT EXISTS (SELECT * FROM dbo.Parameters WHERE Id = @ThreadsId)
-INSERT INTO dbo.Parameters (Id,Number) SELECT @PeriodSecId, 24*3600 WHERE NOT EXISTS (SELECT * FROM dbo.Parameters WHERE Id = @PeriodSecId)
+INSERT INTO dbo.Parameters (Id,Number) SELECT @PeriodHourId, 24 WHERE NOT EXISTS (SELECT * FROM dbo.Parameters WHERE Id = @PeriodHourId)
 INSERT INTO dbo.Parameters (Id,Number) SELECT @HeartbeatPeriodSecId, 60 WHERE NOT EXISTS (SELECT * FROM dbo.Parameters WHERE Id = @HeartbeatPeriodSecId)
 INSERT INTO dbo.Parameters (Id,Number) SELECT @HeartbeatTimeoutSecId, 600 WHERE NOT EXISTS (SELECT * FROM dbo.Parameters WHERE Id = @HeartbeatTimeoutSecId)
 INSERT INTO dbo.Parameters (Id,Char) SELECT name, 'LogEvent' FROM sys.objects WHERE type = 'p' AND name LIKE '%defrag%' AND NOT EXISTS (SELECT * FROM dbo.Parameters WHERE Id = name)
             ";
-                cmd.Parameters.AddWithValue("@IsEnabledId", IsEnabledId);
-                cmd.Parameters.AddWithValue("@ThreadsId", ThreadsId);
-                cmd.Parameters.AddWithValue("@PeriodSecId", PeriodSecId);
-                cmd.Parameters.AddWithValue("@HeartbeatPeriodSecId", HeartbeatPeriodSecId);
-                cmd.Parameters.AddWithValue("@HeartbeatTimeoutSecId", HeartbeatTimeoutSecId);
-                cmd.ExecuteNonQueryAsync(CancellationToken.None).Wait();
-            }
-            catch (SqlException e)
-            {
-                var str = e.ToString();
-                if (str.Contains("login failed", StringComparison.OrdinalIgnoreCase) || str.Contains("dbo.Parameters", StringComparison.OrdinalIgnoreCase))
-                {
-                    _logger.LogWarning($"InitParams exception={str}.");
-                    Thread.Sleep(2000);
-                    goto retry;
-                }
 
-                _logger.LogError($"InitParams exception={str}.");
-                throw;
-            }
+                    cmd.Parameters.AddWithValue("@IsEnabledId", IsEnabledId);
+                    cmd.Parameters.AddWithValue("@ThreadsId", ThreadsId);
+                    cmd.Parameters.AddWithValue("@PeriodHourId", PeriodHourId);
+                    cmd.Parameters.AddWithValue("@HeartbeatPeriodSecId", HeartbeatPeriodSecId);
+                    cmd.Parameters.AddWithValue("@HeartbeatTimeoutSecId", HeartbeatTimeoutSecId);
 
-            _threads = GetThreads();
-            _heartbeatPeriodSec = GetHeartbeatPeriod();
-            _heartbeatTimeoutSec = GetHeartbeatTimeout();
-            _periodSec = GetPeriod();
-            _logger.LogInformation("InitParams completed.");
+                    await cmd.ExecuteNonQueryAsync(CancellationToken.None);
+                });
+        }
+
+        public Task Handle(StorageInitializedNotification notification, CancellationToken cancellationToken)
+        {
+            _storageReady = true;
+            return Task.CompletedTask;
         }
```

Comment:
We will have conflict between long running sql commands and schema upgrades. What would be an appropriate solution for this?